docs/usage/search.md

```diff
@@ -1,3 +1,187 @@
-# :construction: Search
+# Search
 
-Coming soon
+Hide offers powerful search capabilities to help agents navigate and explore projects efficiently. There are three main types of search available:
+
+1. Content Search
+2. File Search
+3. Symbol Search
+
+Let's dive into each of these search types and see how you can leverage them in your projects.
+
+## Content Search
+
+Content search allows you to find specific text patterns within your project files. This is incredibly useful when you need to locate particular code snippets, comments, or any textual content across your entire project.
+
+### Basic Usage
+
+To perform a content search:
+
+=== "curl"
+
+    ```bash
+    curl -X GET "http://localhost:8080/projects/{projectId}/search?type=content&query=your_search_query"
+    ```
+
+=== "python"
+
+    ```python
+    # Coming soon
+    ```
+
+### Advanced Options
+
+Hide's content search supports different search types:
+
+- **Default**: Case-insensitive search
+- **Exact**: Case-sensitive, exact match search
+- **Regex**: Regular expression search
+
+You can specify the search type using query parameters:
+
+=== "curl"
+
+    ```bash
+    # Exact match
+    curl -X GET "http://localhost:8080/projects/{projectId}/search?type=content&query=YourExactPhrase&exact"
+
+    # Regex search
+    curl -X GET "http://localhost:8080/projects/{projectId}/search?type=content&query=Your.*Regex&regex"
+    ```
+
+=== "python"
+
+    ```python
+    # Coming soon
+    ```
+
+### Additional Parameters
+
+Hide's content search also supports additional parameters to refine your search:
+
+- **showHidden**: Include hidden files in the search
```

Comment:
I don't think we allow for that ATM, should we? 
https://github.com/artmoskvin/hide/blob/3421ce0ceb27e46f168279a0668306fdb1162977/pkg/handlers/search_files.go#L138

Reply:
Aren't we adding this param here? https://github.com/hide-org/hide/blob/main/pkg/handlers/utils.go#L34

Reply:
yeah it works
<img width="598" alt="image" src="https://github.com/user-attachments/assets/9a810917-cf02-4d11-bdae-607b0573e6dd">

Reply:
Surprise, heh I suppose my update added a feature. Will remove the arg.